config/channel/distributed/300-eventing-kafka-configmap.yaml

```diff
@@ -14,45 +14,52 @@
 
 apiVersion: v1
 data:
+  version: 1.0.0
   sarama: |
-    Version: 2.0.0 # Kafka Version Compatibility From Sarama's Supported List (Major.Minor.Patch)
-    Admin:
-      Timeout: 10000000000  # 10 seconds
-    Net:
-      KeepAlive: 30000000000  # 30 seconds
-      MaxOpenRequests: 1 # Set to 1 for use with Idempotent Producer
-      TLS:
-        Enable: true
-      SASL:
-        Enable: true
-        Version: 1
-    Metadata:
-      RefreshFrequency: 300000000000  # 5 minutes
-    Consumer:
-      Offsets:
-        AutoCommit:
-          Interval: 5000000000  # 5 seconds
-        Retention: 604800000000000  # 1 week
-    Producer:
-      Idempotent: true  # Must be false for Azure EventHubs
-      RequiredAcks: -1  # -1 = WaitForAll, Most stringent option for "at-least-once" delivery.
+    enableLogging: false
+    config: |
+      Version: 2.0.0 # Kafka Version Compatibility From Sarama's Supported List (Major.Minor.Patch)
+      Admin:
+        Timeout: 10000000000  # 10 seconds
+      Net:
+        KeepAlive: 30000000000  # 30 seconds
+        MaxOpenRequests: 1 # Set to 1 for use with Idempotent Producer
+        TLS:
+          Enable: true
+        SASL:
+          Enable: true
+          Version: 1
+      Metadata:
+        RefreshFrequency: 300000000000  # 5 minutes
+      Consumer:
+        Offsets:
+          AutoCommit:
+            Interval: 5000000000  # 5 seconds
+          Retention: 604800000000000  # 1 week
+      Producer:
+        Idempotent: true  # Must be false for Azure EventHubs
+        RequiredAcks: -1  # -1 = WaitForAll, Most stringent option for "at-least-once" delivery.
   eventing-kafka: |
-    receiver:
-      cpuRequest: 100m
-      memoryRequest: 50Mi
-      replicas: 1
-    dispatcher:
-      cpuRequest: 100m
-      memoryRequest: 50Mi
-      replicas: 1
+    cloudevents:
+      maxIdleConns: 1000
+      maxIdleConnsPerHost: 100
     kafka:
+      authSecretName: kafka-cluster
+      authSecretNamespace: knative-eventing
       brokers: REPLACE_WITH_CLUSTER_URL
-      enableSaramaLogging: false
       topic:
         defaultNumPartitions: 4
         defaultReplicationFactor: 1 # Cannot exceed the number of Kafka Brokers!
         defaultRetentionMillis: 604800000  # 1 week
-      adminType: kafka # One of "kafka", "azure", "custom"
+    channel:
+      distributed:
```

Comment:
I am not 100% sure but I think it might be a better idea to not have the separation of distributed/consolidated in the configmap. In the long run, if we want to have a single channel, users would do another migration of the configmap.

There's pros/cons in both. I would go for not having `consolidated` or `distributed` nested fields and let the implementations ignore the fields they don't use. Just a thought but I am also fine with the approach in the PR.

Reply:
In the end, users can't use both channels at the same time anyway.